src/playground/sample_data_ingestion/utils/sample_data_ingestor/sample_data_ingestor.py

```diff
@@ -0,0 +1,236 @@
+                        copyfileobj(fin, fout)
+                    name = input.split(".gz")[0]
+        # Reads from CSV
+        if ".csv" in name:
+            with open(name, "r") as f:
+                reader = csv.reader(f)
+                fields = next(reader)
+                for row in reader:
+                    entry = {}
+                    for i in range(len(fields)):
+                        entry[fields[i]] = row[i]
+                    if not initial_time and timestamp:
+                        initial_time = entry[timestamp]
+                    dataset.append(entry)
+        # Reads from JSON 
+        elif ".json" in name:
+            with open(name, "r") as f:
+                for line in f:
+                    if not initial_time and timestamp:
+                        temp = loads(line)
+                        initial_time = temp[timestamp]
+                    dataset.append(loads(line))
+        
+        # Removes zipped file
+        if template != name:
+            remove(name)
+
+    # If anomalies wanted to be generated
+    elif anomaly_percentage:
+        dataset = generate_data_trend(input = template, 
+            feature = feature, 
+            timestamp = timestamp, 
+            number = number, 
+            avg_min = avg_min, 
+            avg_max = avg_max, 
+            abs_min = abs_min, 
+            abs_max = abs_max, 
+            anomaly_percentage = anomaly_percentage, 
+            mapping = mapping, 
+            minutes = minutes,
+            current_date = current_date
+        )
+    else:
+         # Generates the specified number of documents
+        for i in range(number):
+            entry = generate_data(template, mapping)
+            if type(entry) is list:
+                for element in entry:
+                    if timestamp:
+                        element[timestamp] = current_date
+                    dataset.append(element)
+            else:
+                entry = generate_data(template, mapping)
+                if timestamp:
+                    entry[timestamp] = current_date
+                dataset.append(entry)
+            current_date += timedelta(minutes = minutes)
+
+
+    # Calls BULK API to ingest documents of size "chunk"
+    i = 0
+    one_week_ago = datetime.now() - timedelta(days = 7)
+    while i < len(dataset):
+        j = i
+        tosend = []
+        while j < min(i + chunk, len(dataset)):
+            tosend.append({"index": {"_index": index_name}})
+
+            # If timestamps were a field type for user provided data, modify time
+            if file_provided and timestamp:
+                one_week_ago += timedelta(minutes = minutes)
+                offset = one_week_ago.strftime("%s")
+                dataset[j][timestamp] = int(offset)
+            tosend.append(dataset[j])
+            j+= 1
+        
+        response = client.bulk(
+            body = tosend
+        )
```

Comment:
let's add validatiion around client.bulk, also think about optimize it based on traffic pattern. One general idea, if the data is super large, can we create mini batch?

Reply:
Done, I validated the request body to make sure it was a list and the list was of even length. As for the traffic pattern optimization, I implemented a feature where the user defines either the max chunk size or byte size of a `BULK` call. My logic takes the lesser of the two configured values to send to OS.